outlines/fsm/guide.py

```diff
@@ -274,3 +247,60 @@ def must_terminate_state(self, state: CFGState) -> bool:
     def copy(self) -> "CFGGuide":
         """Create a copy of the Guide."""
         return CFGGuide(self.cfg_string, self.tokenizer)
+
+
+class Write:
+    def __init__(self, text: str):
+        self.text = text
+
+
+class Generate:
+    def __init__(self, model, allowed_token_ids):
+        self.model = model
+        self.allowed_token_ids = allowed_token_ids
+
+
+def create_states_mapping(fsm, tokenizer):
+    states_mapping = {}
+    for state in fsm.states:
+        states_mapping[state] = {
+            "allowed_token_ids": list(fsm.transitions[state].keys()),
+            "is_final": fsm.is_final(state),
+        }
+    return states_mapping
+
+
+def create_states_mapping_from_fsm(fsm, tokenizer):
+    states_mapping = {}
+    for state in fsm.states:
+        states_mapping[state] = {
+            "allowed_token_ids": list(fsm.transitions[state].keys()),
+            "is_final": fsm.is_final(state),
+        }
+    return states_mapping
```

Comment:
We would want to avoid these kind of "supporters" in the new `RegexGuide` and figure out the way to provide a better interface for it in Guide from `outlines-core`. So, in the experiments here, we need to try using only what's provided to understand what's missing and add it. Here, I'm not sure why do you even need to have `self.states_mapping`?